### PR TITLE
systemd: expand template before start

### PR DESCRIPTION
**Pull request by gsanchietti**
Current installations will be fixed by template expansion fired inside the update event.
This patch will be effective from next upstream netdata update.

NethServer/dev#6528

### CODE REVIEW

File: root/etc/systemd/system/netdata.service.d/cleanup.conf [hunk 1]
```diff
@@ -1,2 +1,5 @@
 [Service]
 ExecStartPre=-/usr/bin/find /var/cache/netdata/ -type f -empty -delete
+ReadWriteDirectories=/etc/netdata/
```
Comment (DavidePrincipi) [inline root/etc/systemd/system/netdata.service.d/cleanup.conf:3]: What is the effect of this directive? How does it alter the service permissions on that directory?

I cannot understand it even after reading the systemd.exec manpage:

> Sets up a new file system namespace for executed processes. These options may be used to limit access a process might have to the main file system hierarchy.

    $ ls -ld /etc/netdata/
    drwxr-xr-x 7 root root 129 Jun 17 16:12 /etc/netdata/

Comment (gsanchietti) [reply root/etc/systemd/system/netdata.service.d/cleanup.conf:3]: The process is started inside a namespace which has read-only access to the filesystem.

See the commit message for more info: https://github.com/NethServer/nethserver-netdata/pull/4/commits

Comment (DavidePrincipi) [reply root/etc/systemd/system/netdata.service.d/cleanup.conf:3]: Ok I guess read-only is triggered by

    ProtectSystem=full

From the manpage:



> ProtectSystem=
>           Takes a boolean argument or "full". If true, mounts the /usr and /boot directories read-only for processes invoked by this unit. If set to "full", the /etc directory is mounted read-only, too.

**Review by DavidePrincipi — APPROVED** [review]

File: root/etc/systemd/system/netdata.service.d/cleanup.conf (hunk 1 above)
Comment (gsanchietti) [reply root/etc/systemd/system/netdata.service.d/cleanup.conf:3]: Yes, it's that one!

**Review by filippocarletti — APPROVED** [review]

**Pull request merged**